File: app/src/main/java/app/mumandroidassignment/lesson5/byjava/question2/DetailActivity.java
package app.mumandroidassignment.lesson5.byjava.question2;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.Locale;

import app.mumandroidassignment.R;
import app.mumandroidassignment.lesson5.bykotlin.question2.bean.Product;

/**
 * Created by CodingHome on 4/8/18.
 */

public class DetailActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        Intent intent = getIntent();
        Product product = (Product) intent.getSerializableExtra("product");
        ImageView icon = findViewById(R.id.im_icon);
        TextView title = findViewById(R.id.tv_title);
        TextView color = findViewById(R.id.tv_color);
        TextView id = findViewById(R.id.tv_id);
        TextView dec = findViewById(R.id.tv_description);

        Glide.with(this)
                .asBitmap()
                .load(product.getImageUrl())
                .into(icon);
        title.setText(product.getTitle());
        color.setText(String.format(Locale.getDefault(), "Color : %s", product.getColor()));
        id.setText("Walmart : " + product.getItemid());
        dec.setText(product.getDesc());
    }
}

File: app/src/main/java/app/mumandroidassignment/lesson5/bykotlin/question2/ui/activity/DetailActivity.kt
package app.mumandroidassignment.lesson5.bykotlin.question2.ui.activity

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.Window
import app.mumandroidassignment.R
import app.mumandroidassignment.lesson5.bykotlin.question2.bean.Product
import app.mumandroidassignment.lesson5.bykotlin.question2.extension.loadImageByGlide
import kotlinx.android.synthetic.main.activity_detail.*

class DetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        val product = intent.getSerializableExtra("product") as Product
        im_icon.loadImageByGlide(product.imageUrl)
        tv_title.text = product.title
        tv_color.text = String.format("Color : %s" + product.color)
        tv_id.text = String.format("Walmart : %s" + product.itemid)
        tv_description.text = product.desc
    }

}
